Packet++/header/TelnetLayer.h

```diff
@@ -0,0 +1,343 @@
+            XDisplayLocation,
+            /// Environment Option RFC1408 https://www.iana.org/go/rfc1408
+            EnvironmentOption,
+            /// Authentication Option RFC2941 https://www.iana.org/go/rfc2941
+            AuthenticationOption,
+            /// Encryption Option RFC2946 https://www.iana.org/go/rfc2946
+            EncryptionOption,
+            /// New Environment Option RFC1572 https://www.iana.org/go/rfc1572
+            NewEnvironmentOption,
+            /// TN3270E RFC2355 https://www.iana.org/go/rfc2355
+            TN3270E,
+            /// X Server Authentication
+            XAuth,
+            /// Charset RFC2066 https://www.iana.org/go/rfc2066
+            Charset,
+            /// RSP, Telnet Remote Serial Port
+            TelnetRemoteSerialPort,
+            /// Com Port Control Option RFC2217 https://www.iana.org/go/rfc2217
+            ComPortControlOption,
+            /// Telnet Suppress Local Echo
+            TelnetSuppressLocalEcho,
+            /// Telnet Start TLS
+            TelnetStartTLS,
+            /// Kermit RFC2840 https://www.iana.org/go/rfc2840
+            Kermit,
+            /// Send URL
+            SendURL,
+            /// Forward X Server
+            ForwardX,
+
+            /// Telnet Option Pragma Logon
+            TelOptPragmaLogon = 138,
+            /// Telnet Option SSPI Logon
+            TelOptSSPILogon,
+            /// Telnet Option Pragma Heartbeat
+            TelOptPragmaHeartbeat,
+
+            /// Extended option list
+            ExtendedOptions = 255
+
+        };
+
+        /**
+         * A constructor that creates the layer from an existing packet raw data
+         * @param[in] data A pointer to the raw data
+         * @param[in] dataLen Size of the data in bytes
+         * @param[in] prevLayer A pointer to the previous layer
+         * @param[in] packet A pointer to the Packet instance where layer will be stored in
+         */
+        TelnetLayer(uint8_t *data, size_t dataLen, Layer *prevLayer, Packet *packet) : Layer(data, dataLen, prevLayer, packet)
+        {
+            isData = false;
+            m_Protocol = Telnet;
+            parseTelnetFields();
+        };
+
+        /**
+         * Get the Telnet data as readable string
+         * @param[in] removeEscapeCharacters Whether non-alphanumerical characters should be removed or not
+         * @return Full payload as readable string, empty if Telnet packet contains control commands/options.
+         */
+        std::string getDataAsString(bool removeEscapeCharacters = true);
+
+        /// Return the number of detected Telnet Commands
+        uint16_t getNumberOfCommands();
+
+        /**
+         * Get the command of the given index
+         * @param[in] index Index to requested value
+         * @return Command of the given index, TelnetCommandInternalError if the requested index is larger than number of commands
+         */
+        TelnetCommands getCommand(size_t index);
```

Comment:
I know that telnet data isn't similar to TLV options (Type-Length-Value), but maybe we can borrow some ideas from that interface. You can look at [`TLVData.h`](https://github.com/seladb/PcapPlusPlus/blob/master/Packet++/header/TLVData.h) and a few layers that implement it such as [`TcpLayer`](https://github.com/seladb/PcapPlusPlus/blob/master/Packet++/header/TcpLayer.h), [`IPv4Layer`](https://github.com/seladb/PcapPlusPlus/blob/master/Packet++/header/IPv4Layer.h), and more - and think if we want a somehow similar interface, for example:
- Create a struct/class for every command
- Build an interface that includes:
  - `getFirstCommand()`
  - `getNextCommand(prevCommand)`
  - `Command getCommand(commandType)`
  - `getCommandCount()`